### PR TITLE
[Feature] Support Create Dynamic Partition Immediately FirstTime Without Wating Schedule.

### PR DESCRIPTION
**Pull request by HappenLee**
## Proposed changes

issue:#5208

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation Update (if none of the other choices apply)
- [] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have create an issue on (Fix #ISSUE), and have described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] If this change need a document change, I have updated the document
- [x] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/clone/DynamicPartitionScheduler.java [hunk 1]
```diff
@@ -17,6 +17,7 @@
 
 package org.apache.doris.clone;
 
+import com.google.common.collect.Lists;
```
Comment (EmmyMiao87) [inline fe/fe-core/src/main/java/org/apache/doris/clone/DynamicPartitionScheduler.java:20]: please attention import order

Comment (HappenLee) [reply fe/fe-core/src/main/java/org/apache/doris/clone/DynamicPartitionScheduler.java:20]: OK

**Review by pengxiangyu — APPROVED** [review]

**Review by pengxiangyu — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/clone/DynamicPartitionScheduler.java [hunk 2]
```diff
@@ -248,15 +254,14 @@ public void createOrUpdateRuntimeInfo(String tableName, String key, String value
         return dropPartitionClauses;
     }
 
-    private void executeDynamicPartition() {
-        Iterator<Pair<Long, Long>> iterator = dynamicPartitionTableInfo.iterator();
-        while (iterator.hasNext()) {
-            Pair<Long, Long> tableInfo = iterator.next();
+    private void executeDynamicPartition(Iterator<Pair<Long, Long>> tableInfoIterator) {
```
Comment (yangzhg) [inline fe/fe-core/src/main/java/org/apache/doris/clone/DynamicPartitionScheduler.java:257]: I think use List is better

**Review by EmmyMiao87 — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/common/util/DynamicPartitionUtil.java [hunk 3]
```diff
@@ -240,6 +240,7 @@ public static void registerOrRemoveDynamicPartitionTable(long dbId, OlapTable ol
         if (olapTable.getTableProperty() != null
                 && olapTable.getTableProperty().getDynamicPartitionProperty() != null) {
             if (olapTable.getTableProperty().getDynamicPartitionProperty().getEnable()) {
+                Catalog.getCurrentCatalog().getDynamicPartitionScheduler().executeDynamicPartitionFirstTime(dbId, olapTable.getId());
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/common/util/DynamicPartitionUtil.java:243]: `registerOrRemoveDynamicPartitionTable()` is also called in `replayCreateTable()`.
And in "replay" process, we should not call `executeDynamicPartition()`

**Review by morningman — CHANGES_REQUESTED** [review]

**Review by morningman — DISMISSED** [review]
LGTM

**Comment by morningman** [comment]
Fix ut

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**